src/d3a/models/strategy/__init__.py

```diff
@@ -646,14 +646,16 @@ def update_offer_rates(self, market: "OneSidedMarket", updated_rate: float) -> N
             return
 
         for offer, iterated_market_id in self.offers.open.items():
+            updated_price = round(offer.energy * updated_rate, 10)
+            if offer.price == updated_price:
```

Comment:
Minor, but since we are talking about floating points, and we know how well does Python handle floating point precision (not), I would feel much safer if you used math.isclose here. Thanks!